client/app/dbaas/logs/detail/inputs/logs-inputs.service.js

```diff
@@ -0,0 +1,256 @@
+     * @param {any} inputId
+     * @returns promise which will resolve with the operation object
+     * @memberof LogsInputsService
+     */
+    restartInput (serviceName, input) {
+        return this.InputsApiLexiService.restart({ serviceName, inputId: input.info.inputId })
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation.data, "logs_inputs_restart_success", { inputTitle: input.info.title });
+            })
+            .catch(err => {
+                this._resetAllCache();
+                return this._handleError("logs_inputs_restart_error", err, { inputTitle: input.info.title });
+            });
+    }
+
+    /**
+     * start an input
+     *
+     * @param {any} serviceName
+     * @param {any} inputId
+     * @returns promise which will resolve with the operation object
+     * @memberof LogsInputsService
+     */
+    startInput (serviceName, input) {
+        return this.InputsApiLexiService.start({ serviceName, inputId: input.info.inputId })
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation.data, "logs_inputs_start_success", { inputTitle: input.info.title });
+            })
+            .catch(err => {
+                this._resetAllCache();
+                return this._handleError("logs_inputs_start_error", err, { inputTitle: input.info.title });
+            });
+    }
+
+    /**
+     * stop an input
+     *
+     * @param {any} serviceName
+     * @param {any} inputId
+     * @returns promise which will resolve with the operation object
+     * @memberof LogsInputsService
+     */
+    stopInput (serviceName, input) {
+        return this.InputsApiLexiService.end({ serviceName, inputId: input.info.inputId })
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation.data, "logs_inputs_stop_success", { inputTitle: input.info.title });
+            })
+            .catch(err => {
+                this._resetAllCache();
+                return this._handleError("logs_inputs_stop_error", err, { inputTitle: input.info.title });
+            });
+    }
+
+    /**
+     * transforms the input by adding some additional information
+     *
+     * @param {any} input
+     * @returns the transformed input
+     * @memberof LogsInputsService
+     */
+    transformInput (input) {
+        input.info.engine.software = [input.info.engine.name, input.info.engine.version].join(" ");
+        input.actionsMap = input.actions.reduce((actions, action) => {
+            actions[action.type] = action.isAllowed;
+            return actions;
+        }, {});
+        input.info.state = input.info.status === this.LogsInputsConstant.status.PROCESSING ? this.LogsInputsConstant.state.PROCESSING :
```

Comment:
Use a switch case if you can. 

Or use some vars to simplify this code.

Reply:
Variables are used to simplify this code